lib/rust/ensogl/core/src/control/io/mouse.rs

```diff
@@ -9,88 +9,46 @@ use crate::system::web;
 
 use std::cell::RefCell;
 use std::rc::Rc;
-use wasm_bindgen::prelude::Closure;
-use wasm_bindgen::JsCast;
-use wasm_bindgen::JsValue;
+use web::Closure;
+use web::JsCast;
+use web::JsValue;
 
 pub use crate::frp::io::mouse::*;
 pub use event::*;
 
 
 
-// =======================
-// === EventDispatcher ===
-// =======================
-
-// TODO: Consider merging this implementation with crate::control::callback::* ones.
-
-/// Shared event dispatcher.
-#[derive(Debug, CloneRef, Derivative)]
-#[derivative(Clone(bound = ""))]
-#[derivative(Default(bound = ""))]
-pub struct EventDispatcher<T> {
-    rc: Rc<RefCell<callback::Registry1<T>>>,
-}
-
-impl<T> EventDispatcher<T> {
-    /// Adds a new callback.
-    pub fn add<F: FnMut(&T) + 'static>(&self, f: F) -> callback::Handle {
-        self.rc.borrow_mut().add(f)
-    }
-
-    /// Dispatches event to all callbacks.
-    pub fn dispatch(&self, t: &T) {
-        self.rc.borrow_mut().run_all(t);
-    }
-}
-
-
-
 // ====================
 // === MouseManager ===
 // ====================
 
-/// An utility which registers JavaScript handlers for mouse events and translates them to Rust
+/// A utility which registers JavaScript handlers for mouse events and translates them to Rust
```

Comment:
This line is not a correct sentence because of a missing full stop :wink:

Reply:
Because the sentence ends in the second line ;)